zipline/protocol.py

```diff
@@ -194,6 +200,87 @@ def current_portfolio_weights(self):
         return position_values / self.portfolio_value
 
 
+class AlgorithmPortfolio(Portfolio):
```

Comment:
No reason other than that I felt I these were pretty radical additions to add to the (previously simple) `Portfolio` class, so I thought it warranted their own sublclass. I wouldn't be opposed to keeping it all in `Portfolio` though if you think that makes more sense.